lib/Yield/Support/GraphLayout/SugiyamaStyle/NodeRanking.cpp

```diff
@@ -133,11 +130,12 @@ static bool areDisjoint(NodeView A, NodeView B, size_t MaxDepth = -1) {
 /// rejoin again.
 constexpr auto DDFS = RankingStrategy::DisjointDepthFirstSearch;
 template<>
-RankContainer rankNodes<DDFS>(InternalGraph &Graph, int64_t DiamondBound) {
+RankContainer
+rankNodes<DDFS>(InternalNode *GraphEntryPoint, int64_t DiamondBound) {
   RankContainer Ranks;
 
   std::vector<NodeView> Stack;
-  for (auto *Node : Graph.nodes()) {
+  for (auto *Node : llvm::depth_first(GraphEntryPoint)) {
     if (!Node->hasPredecessors() && !Ranks.contains(Node)) {
       Stack.emplace_back(Node);
```

Comment:
This is weird. Now you're doing a DFS but you only do stuff when `!Node->hasPredecessors()`. So in my understanding you'd be only doing stuff when `Node` is `GraphEntryPoint`. At this point why doing the DFS in the first place?

Reply:
Ah, I didn't really think about it. My thought process was:
- this just wants a list of nodes, while all the other ranking algorithms just want an entry.
- but wait! There's an easy way to get list of nodes from the entry. And it's not like it cares about the order they are in.

To summarize, does just pulling this preparation step outside the loop make sense to you?

Reply:
TL;DR: this ranking algorithm probably wants to be redesigned, to be consistent with the rest, but I'd guess today is not the day.

Reply:
> this ranking algorithm probably wants to be redesigned, to be consistent with the rest, but I'd guess today is not the day.

That's definitely true, but I'd not focus on it in this PR.